assessments/0010-etcd/etcd-implementation.md

```diff
@@ -0,0 +1,198 @@
+
+The top-level documentation recommendations for this project are:
+
+- Complete and update instructional documentation:
+    - Describe etcd's user roles (personas).
+    - Document Kubernetes and Linux installation procedures.
+    - Update and clarify quick start and new user workflows.
+    - Document key tasks as procedures rather than tutorials.
+    - Clarify which user roles (personas) each task is for.
+    - Going forward, write release notes for each major and minor release.
+- Reorganize the documentation.
+    - Move *project* documentation (documentation for OSS contributors) to the etcd GitHub repo. Move *product* documentation (documentation for those who use the etcd software, in any capacity) to the technical documentation in the website repo.
+    - Make instructional documentation (tasks and procedures) easier to find by putting them in their own section (like the tutorials currently are). Rename the "Tutorials" section to "Tasks". The tutorials in the current doc are actually granular tasks.
+    - Rename the "Learning" section to something more descriptive such as "Technical Overview", and move it near the top of the table of contents.
+    - Consolidate all reference information (APIs and Glossary) to a reference-specific section.
+    - Remove information about old software revisions from the current version's docs if it's not necessary for a significant segment of users. This includes upgrade procedures and performance benchmarks.
+
+
+# Implementation: Complete and update documentation
+
+In general, ensure that task-based documentation is complete and accurate. Write common tasks as step-by-step instructions in individual descriptions, each with the goal of walking the user through completion of the task. This is distinct from tutorials, which are designed to teach the user how to accomplish a more complex use case. 
+
+Currently, the "Tutorials" section in the documentation contains tasks that should be documented as such. 
+
+For example, the [How to access etcd][access-tutorial] tutorial should be two separate tasks: "Writing to etcd on the CLI" and "Reading from etcd on the CLI".
+
+Name the tasks using gerund phrases ("-ing" form, like "Reading ..." and "Writing ..."). Eliminate "How to", which clutters the heading with a redundant phrase that the user must skip over. 
+
+## Describe etcd's user roles
+
+In an "Overview" or "Start here" page, outline etcd's user roles or personas. These probably include:
+- *Evaluator*: Someone trying to determine whether etcd is appropriate for their product, project, or organization.
+- *Admin or Operator*: Someonereponsible for setting up and maintaining a production etcd service.
```

Comment:
It's worth splitting Admin into two sub-roles:  (a) Admin using Etcd embedded in Kubernetes (b) admin using Etcd on its own.

Reply:
That's an excellent suggestion. I'll need to collaborate with a contributor who understands the differences between the sub-roles in order to update the implementation plan.